source/server/listener_manager_impl.h

```diff
@@ -268,7 +269,9 @@ class ListenerImpl : public Network::ListenerConfig,
   const envoy::api::v2::core::Metadata& listenerMetadata() const override {
     return config_.metadata();
   };
+  SystemTimeSource& systemTimeSource() override { return parent_.system_time_source_; }
   void ensureSocketOptions() {
+    last_updated_ = systemTimeSource().currentTime();
```

Comment:
Yeah, definitely wondering what the rationale behind when to capture the `last_updated_` is here.

Reply:
Good catch -- I was conflating user-initiated config changes with internal mutations of configs. I'll get rid of this and the one above.